sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_telemetry.c

```diff
@@ -76,213 +47,287 @@ static void test_az_iot_hub_client_telemetry_publish_topic_get_NULL_client_fails
       AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY);
 }
 
-static void test_az_iot_hub_client_telemetry_publish_topic_get_with_options_module_id_with_params_succeed(
+static void
+test_az_iot_hub_client_telemetry_publish_topic_get_with_options_module_id_with_props_succeed(
     void** state)
 {
   (void)state;
 
-  uint8_t mqtt_topic_buf[sizeof(g_test_correct_topic_with_options_module_id_with_params) - 1];
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  options.module_id = test_module_id;
+
+  az_iot_hub_client client;
+  assert_int_equal(
+      az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options), AZ_OK);
+
+  az_iot_hub_client_properties props;
+  assert_int_equal(az_iot_hub_client_properties_init(&props, test_props), AZ_OK);
 
+  uint8_t mqtt_topic_buf[sizeof(g_test_correct_topic_with_options_module_id_with_props) - 1];
   az_span mqtt_topic = az_span_init(mqtt_topic_buf, 0, _az_COUNTOF(mqtt_topic_buf));
 
   assert_int_equal(
-      az_iot_hub_client_telemetry_publish_topic_get(
-          &g_test_valid_client_with_options_module_id, &g_test_params, mqtt_topic, &mqtt_topic),
+      az_iot_hub_client_telemetry_publish_topic_get(&client, &props, mqtt_topic, &mqtt_topic),
       AZ_OK);
   assert_memory_equal(
-      g_test_correct_topic_with_options_module_id_with_params,
+      g_test_correct_topic_with_options_module_id_with_props,
       (char*)az_span_ptr(mqtt_topic),
-      sizeof(g_test_correct_topic_with_options_module_id_with_params) - 1);
+      sizeof(g_test_correct_topic_with_options_module_id_with_props) - 1);
 }
 
-static void test_az_iot_hub_client_telemetry_publish_topic_get_with_options_module_id_with_params_small_buffer_fails(
+static void
+test_az_iot_hub_client_telemetry_publish_topic_get_with_options_module_id_with_props_small_buffer_fails(
     void** state)
 {
   (void)state;
 
-  uint8_t mqtt_topic_buf[sizeof(g_test_correct_topic_with_options_module_id_with_params) - 2];
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  options.module_id = test_module_id;
+
+  az_iot_hub_client client;
+  assert_int_equal(
+      az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options), AZ_OK);
 
+  az_iot_hub_client_properties props;
+  assert_int_equal(az_iot_hub_client_properties_init(&props, test_props), AZ_OK);
+
+  uint8_t mqtt_topic_buf[sizeof(g_test_correct_topic_with_options_module_id_with_props) - 2];
   az_span mqtt_topic = az_span_init(mqtt_topic_buf, 0, _az_COUNTOF(mqtt_topic_buf));
 
   assert_int_equal(
-      az_iot_hub_client_telemetry_publish_topic_get(
-          &g_test_valid_client_with_options_module_id, &g_test_params, mqtt_topic, &mqtt_topic),
+      az_iot_hub_client_telemetry_publish_topic_get(&client, &props, mqtt_topic, &mqtt_topic),
       AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY);
 }
 
 int test_iot_hub_telemetry()
 {
+#ifndef NO_PRECONDITION_CHECKING
```

Comment:
ugh, not a fan of double negatives...

Reply:
yea this isn't our primitive this is a core option. it is what it is :)